app/src/main/java/com/sdpteam/connectout/event/viewer/EventActivity.java

```diff
@@ -83,29 +92,37 @@ private void initEventView() {
         ImageButton chatBtn = findViewById(R.id.event_chat_btn);
 
         participationBtn.setOnClickListener(v -> {
-            eventViewModel.toggleParticipation(currentUserId);
+            eventViewModel.toggleParticipation(currentUserId, profileViewModel,
+                    x -> impossibleRegistrationToast(x), (p,e) -> isRegistrationPossible(p,e),
+                    e -> {final Intent intent = new Intent(this, SetEventRestrictionsActivity.class);
+                        intent.putExtra(PASSED_ID_KEY, e.getId());
+                        startActivity(intent);
+            });
```

Comment:
I would extract a method for this since the method has already lot of lines, but let's do that during the merge